📖 Add Documentation for CRD and Operator Scope (#3804)
kubernetes-sigs/kubebuilder: docs/book/src/reference/operator-scope.md

> # Operator Scope

Reviewer: ```suggestion
# Manager Scope
```

> This section dives into the two crucial concepts of Custom Resource Definitions (CRDs) and Operators, specifically their respective scope and how both impact resource management in your Kubernetes environment.

Reviewer: ```suggestion
```

I think we do not follow this standard in the Kubebuilder doc we do not need this header.

> This section dives into the two crucial concepts of Custom Resource Definitions (CRDs) and Operators, specifically their respective scope and how both impact resource management in your Kubernetes environment.

Reviewer: ```suggestion
```

> # Overview

Reviewer: ```suggestion
```

> In the world of Kubernetes operators, scope defines the reach of an operator's management capabilities. This essentially means deciding which resources across your Kubernetes cluster the operator can watch and manage. Here's a breakdown of the two main options:

Reviewer: ```suggestion
In the world of Kubernetes solutions, scope defines the reach of an projects's management capabilities. This essentially means deciding which resources across your Kubernetes cluster the manager can watch and manage. Here's a breakdown of the two main options:
```

- **Namespace-scoped**: Only watches and manages resources within a single namespace.
- **Cluster-scoped**: Watches and manages resources across the entire cluster.

## Choosing the Right Scope:

- **Cluster-scoped operators**: Ideal for managing resources that can be created in any namespace, like certificate management or centralized configuration.
- **Namespace-scoped operators**: Suitable for scenarios like:
- **Flexible deployment**: Allows for independent upgrades and isolation of failures within a namespace.
- **Differing API definitions**: Enables specific configurations for individual namespaces.

## Defaults and Considerations:

> - The operator-sdk init command by default creates a cluster-scoped operator.

Reviewer: ```suggestion
- The kubebuilder init command by default creates a cluster-scoped operator.
```

```go 
...
watchNamespace, err := getWatchNamespace()
if err != nil {
    setupLog.Error(err, "unable to get WatchNamespace, " +
        "the manager will watch and manage resources in all Namespaces")
}

options := ctrl.Options{
    Scheme:             scheme,
    MetricsBindAddress: metricsAddr,
    Port:               9443,
    LeaderElection:     enableLeaderElection,
    LeaderElectionID:   "f1c5ece8.example.com",
    Namespace:          watchNamespace, // namespaced-scope when the value is not an empty string
}

// Add support for MultiNamespace set in WATCH_NAMESPACE (e.g ns1,ns2)
if strings.Contains(watchNamespace, ",") {
    setupLog.Info("manager set up with multiple namespaces", "namespaces", watchNamespace)
    // configure cluster-scoped with MultiNamespacedCacheBuilder
    options.Namespace = ""
    options.NewCache = cache.MultiNamespacedCacheBuilder(strings.Split(watchNamespace, ","))
}
...
```
- Define the environment variable in the config/manager/manager.yaml:

```yaml
...
    env:
      - name: WATCH_NAMESPACE
        value: "ns1,ns2"
  terminationGracePeriodSeconds: 10
...
```
## Updating your CSV’s installModes

> If your operator is integrated with OLM, you will want to update your CSV base’s spec.installModes list to support the desired namespacing requirements. Support for multiple types of namespacing is allowed, so supporting multiple install modes in a CSV is permitted. After doing so, update your bundle or package manifests by following the linked guides.

Reviewer: This stuff is not valid for us.
It is valid only for Operator-SDK
